src/com/google/enterprise/adaptor/database/UniqueKey.java

```diff
@@ -239,9 +243,8 @@ String makeUniqueId(ResultSet rs, boolean encode) throws SQLException {
           part = "" + rs.getLong(name);
           break;
         default:
-          String errmsg = "Invalid type '" + types.get(name) + "' for '"
-              + name + "'. Valid types are: int, string, timestamp, date"
-              + ", time and long.";
+          String errmsg = "Unknown column type for '" + name + "'. "
```

Comment:
I think my two comments here didn't go together. With UNKNOWN, this was a config error, but now it's back to an AssertionError (it shouldn't happen, because every enum value is accounted for).